drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/DataBuilder.java

```diff
@@ -97,92 +112,104 @@ public void build() {
-                        }
+                        setValue(cell, dataType);
                     }
                 }
                 sourceColumnIndex++;
                 targetColumnIndex++;
             }
             rowCount++;
         }
-    }
 
-    private String fixStringValue(DTCellValue52 cell) {
-        if (cell.getStringValue().length() > 2 && cell.getStringValue().startsWith("\"") && cell.getStringValue().endsWith("\"")) {
-            return cell.getStringValue().substring(1, cell.getStringValue().length() - 1);
-        } else {
-            return cell.getStringValue();
+        private void setValue(final DTCellValue52 cell,
+                              final DataType.DataTypes dataType) {
+            switch (dataType) {
+
+                case STRING:
+
+                    if (isTheRealCellValueString(sourceColumnIndex) && cell.getStringValue() != null) {
+
+                        if (cell.getStringValue() != null && !cell.getStringValue().isEmpty()) {
+
+                            xlsRow.createCell(targetColumnIndex)
+                                    .setCellValue(String.format("\"%s\"", fixStringValue(cell)));
+                        }
+                    } else {
+                        /*
+                         * Values from lists of dates or enumerations are stored as Strings.
+                         * For this reason we check the real data type and drop the surrounding "" if the real value was not String.
+                         */
+                        if (Objects.equals(DataType.DataTypes.STRING, cell.getDataType())) {
+
+                            xlsRow.createCell(targetColumnIndex)
+                                    .setCellValue(cell.getStringValue());
+                        } else {
+
+                            setValue(cell,
+                                     cell.getDataType());
+                        }
+                    }
+
+                    break;
+                case NUMERIC:
+                case NUMERIC_BIGDECIMAL:
+                case NUMERIC_BIGINTEGER:
+                case NUMERIC_BYTE:
+                case NUMERIC_DOUBLE:
+                case NUMERIC_FLOAT:
+                case NUMERIC_INTEGER:
+                case NUMERIC_LONG:
+                case NUMERIC_SHORT:
+                    final Number numericValue = cell.getNumericValue();
+                    if (numericValue != null) {
+                        xlsRow.createCell(targetColumnIndex)
+                                .setCellValue(numericValue.toString());
+                    }
+                    break;
+                case DATE:
+                    final Date dateValue = cell.getDateValue();
+                    if (dateValue != null) {
+
+                        final SimpleDateFormat formatter = new SimpleDateFormat(getDateFormat());
+
+                        xlsRow.createCell(targetColumnIndex)
+                                .setCellValue(formatter.format(dateValue));
```

Comment:
We need `.setCellValue(String.format("\"%s\"", formatter.format(dateValue)));` otherwise validation of the converted fails

Reply:
Did this and changed the BOOLEAN to set the text to true/false instead of the XLS boolean cell. I think the boolean cell shows as 0/1 in some editors and true/false in others.